packages/flutter/lib/src/cupertino/tableview.dart

```diff
@@ -0,0 +1,485 @@
+              setState(() {
+                _lastItemSize = size;
+              });
+            });
+          }
+        },
+        child: cellWithDivider,
+      );
+    } else {
+      return cellWithDivider;
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    // Visual list of cells.
+    final Widget sliverList = SliverList(
+      delegate: SliverChildBuilderDelegate(
+        _cellDecorator,
+        childCount: widget.plainChildCount,
+      ),
+    );
+
+    // Visual list of cells + extra dividers at the bottom.
+    final Widget scrollingList = CustomScrollView(
+      controller: widget.scrollController,
+      slivers: <Widget>[]
+        ..add(sliverList)
+        ..add(CupertinoTableViewExtraDividers(
+          cellHeight: _lastItemSize?.height,
+        )),
+    );
+
+    // Apply a background color beneath the cells in the table.
+    return Container(
+      color: widget.backgroundColor,
+      child: scrollingList,
+    );
+  }
+}
+
+// TODO(mattcarroll): replace Painter with a Decoration
+class _CupertinoTableViewDivider extends StatelessWidget {
+  const _CupertinoTableViewDivider({
+    this.child,
+  });
+
+  final Widget child;
+
+  @override
+  Widget build(BuildContext context) {
+    return CustomPaint(
+      foregroundPainter: _CupertinoTableViewDividerPainter(),
+      child: child,
+    );
+  }
+}
+
+class _CupertinoTableViewDividerPainter extends CustomPainter {
+  _CupertinoTableViewDividerPainter({
+    this.dividerThickness = kDefaultDividerHeight,
+    this.dividerInset = kDefaultCellInset,
+    this.dividerColor = CupertinoColors.lightBackgroundGray,
+  }) : _dividerPaint = Paint()
+    ..color = dividerColor
+    ..style = PaintingStyle.fill;
+
+  final double dividerThickness;
+  final double dividerInset;
+  final Color dividerColor;
+  final Paint _dividerPaint;
+
+  @override
+  void paint(Canvas canvas, Size size) {
+    canvas.drawRect(
+      Rect.fromLTRB(
+        dividerInset,
+        size.height - dividerThickness,
+        size.width,
+        size.height,
+      ),
+      _dividerPaint,
+    );
+  }
+
+  @override
+  bool shouldRepaint(_CupertinoTableViewDividerPainter oldPainter) {
+    return  dividerThickness != oldPainter.dividerThickness ||
+        dividerColor != oldPainter.dividerColor ||
+        dividerInset != oldPainter.dividerInset;
+  }
+
+}
+
+/// Widget that renders a [Sliver] that draws extra dividers after the last cell
+/// in a [CupertinoTableView].
+///
+/// No dividers will be drawn if [CupertinoTableView] has enough cells to scroll,
+/// as per iOS behavior.
+///
+/// The gap between extra dividers is equal to the height of the last cell in
+/// the [CupertinoTableView], as per iOS behavior.
+@visibleForTesting
+class CupertinoTableViewExtraDividers extends RenderObjectWidget {
+  const CupertinoTableViewExtraDividers({
+    double cellHeight,
+  }) : cellHeight = cellHeight ?? kDefaultTableViewCellHeight;
```

Comment:
So the first frame will always flash? We should fix this.

Reply:
There's no flash as far as I've seen. But if you have an alternative approach that isn't hideous then I'm happy to hear it.  I couldn't find an obvious one.

Reply:
I changed cellHeight to _CellHeightController with ChangeNotifier which allowed me to get rid of a call to `setState()`.  So I think the problem is solved.